app/docker/views/containers/create/createContainerController.js

```diff
@@ -619,62 +581,191 @@ function ($q, $scope, $state, $timeout, $transition$, $filter, Container, Contai
     return true;
   }
 
-  $scope.create = function () {
-    confirmCreateContainer()
-    .then(function success(confirm) {
-      if (!confirm) {
-        return false;
+
+  function create() {
+    var oldContainer = null;
+
+    $scope.state.actionInProgress = true;
+    HttpRequestHelper.setPortainerAgentTargetHeader($scope.formValues.NodeName);
+    return findCurrentContainer()
+      .then(confirmCreateContainer)
+      .then(startCreationProcess)
+      .catch(notifyOnError)
+      .finally(final);
+
+    function final() {
+      $scope.state.actionInProgress = false;
+    }
+
+    function findCurrentContainer() {
+      return Container.query({ all: 1, filters: { name: ['^/' + $scope.config.name + '$'] } })
+        .$promise
+        .then(function onQuerySuccess(containers) {
+          if (!containers.length) {
+            return;
+          }
+          oldContainer = containers[0];
+          return oldContainer;
+        })
+        .catch(notifyOnError);
+
+      function notifyOnError(err) {
+        Notifications.error('Failure', err, 'Unable to retrieve containers');
       }
+    }
 
-      var accessControlData = $scope.formValues.AccessControlData;
-      var userDetails = Authentication.getUserDetails();
-      var isAdmin = userDetails.role === 1;
+    function startCreationProcess(confirmed) {
+      if (!confirmed) {
+        return $q.when();
+      }
+      if (!validateAccessControl()) {
+        return $q.reject({ msg: 'Not authorized' });
```

Comment:
Why do you raise this kind of error here? `validateAccessControl` is supposed to simply validate the form fields.